### PR TITLE
[receiver/sqlserver] Add computer name resource attribute to relevant metrics

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

The computer name resource attribute is relevant for all metrics coming from the SQL server receiver when directly connected to the DB. This change records the resource attribute with all remaining metrics that it wasn't recorded with yet.

**Link to tracking Issue:** <Issue number if applicable>
This was discovered at the same time as https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35036, but I split out the changes to hopefully make them easier to review.

**Testing:** <Describe what testing was performed and which tests were added.>
Tests have been updated and are passing.

### CODE REVIEW

File: .chloggen/sqlserver_computer_rattr.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: sqlserverreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Add computer name resource attribute to relevant metrics
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: []
```
Comment (crobert-1) [inline .chloggen/sqlserver_computer_rattr.yaml:13]:
```suggestion
issues: [35040]
```

**Review by djaglowski — APPROVED** [review]

File: .chloggen/sqlserver_computer_rattr.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
```
[the remark this answers is not in the document]
Comment (atoulme) [reply .chloggen/sqlserver_computer_rattr.yaml:4]: looks ok to me

**Review by atoulme — APPROVED** [review]
LGTM, please resolve the conflict

**Comment by crobert-1** [comment]
I've filed https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35226 for the failing integration test, it's unrelated to this change.

_Adding `ready to merge` as this PR has necessary approvals._

**Review by pjanotti — APPROVED** [review]

**Pull request merged**